### PR TITLE
Change method for loading energies from Gaussian log files to regex search to avoid errors 

### PR DESCRIPTION
**Pull request by donerancl**
### Motivation or Problem
Sometimes, in  gaussian log files, the line that contains the energy has extra characters (more than one line printed in the same line) that prevent the load_energy function in arkane/ess/gaussian.py from parsing the energy. (e.g., it will try to use an "=" as the energy)

### Description of Changes
Before: find the line that has "SCF Done:", split it, take the 4th index as the energy

After: find the line that has "SCF Done:", find the substring that matches "SCF Done: E(...) = {energyfloat}, split the substring, and take the last item

the same can be done for E2, MP2, E(CORR), CCSD(T), CBS-QB3 (0 K), E(CBS-QB3)=, CBS-4 (0 K)=, G3(0 K), G3 Energy=,G4(0 K), G4 Energy=, G4MP2(0 K), G4MP2 Energy= in the same function

### Testing
I plan to try the load_energy function on many gaussian log files, some which have extra characters in the line containing the energy

### Reviewer Tips
Suggestions for verifying that this PR works or other notes for the reviewer.

### CODE REVIEW

**Comment by JacksonBurns** [comment]
I know this isn't ready yet, but when you are good for review please [squash these commits](https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together). Thanks!

**Comment by donerancl** [comment]
I think this is ready. I was able to parse thousands of gaussian log files without the errors from the issue I described above.

File: arkane/ess/gaussian.py [hunk 1]
```diff
@@ -376,6 +375,12 @@ def load_energy(self, zpe_scale_factor=1.):
                             elect_energy_source = 'HF'
                     except ValueError:
                         pass
+                elif 'Energy=' in line:
+                    # for xtb
+                    e_elect = float(line.split()[1]) * constants.E_h * constants.Na
+                elif 'Energy=' in line:
+                    # for xtb
+                    e_elect = float(line.split()[1]) * constants.E_h * constants.Na
```
Comment (JacksonBurns) [inline arkane/ess/gaussian.py:383]: This is a duplicate of the above, I think should be removed.

**Review by JacksonBurns — CHANGES_REQUESTED** [review]
Thanks for squashing! The new regex seems to work, as it isn't failing any such tests.

This is failing a couple of the other unit tests - it seems that the new `Energy=` test is picking up some non-numeric text in some cases, like in this unit test: https://github.com/ReactionMechanismGenerator/RMG-Py/actions/runs/7732973138/job/21084096743?pr=2595#step:10:2616
which is attempting to read this Gaussian log file: https://github.com/ReactionMechanismGenerator/RMG-Py/blob/b53f63bf7826440c1e410e311136fa606167c332/arkane/data/gaussian/oxygen.log

It picks up either "change" or "forces"

File: arkane/ess/gaussian.py (hunk 1 above)
Comment (donerancl) [reply arkane/ess/gaussian.py]: Yes, I must have added these in by accident, I think they are not needed.

**Review by JacksonBurns — DISMISSED** [review]
Thanks for the quick fix! LGTM, thanks for the improvement.

**Comment by JacksonBurns** [comment]
Rebased and enabled auto-merge, so this should go in on its own tonight.

**Comment by JacksonBurns** [comment]
Arg this is blocked by CI issues, apologies. Addressing now, will circle back this...

**Comment by JacksonBurns** [comment]
Issue found, blocked by #2608

**Comment by JacksonBurns** [comment]
Now blocked by https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2658

**Comment by JacksonBurns** [comment]
@hwpang please review as well

File: arkane/ess/gaussian.py [hunk 2]
```diff
@@ -309,9 +309,8 @@ def load_energy(self, zpe_scale_factor=1.):
         with open(self.path, 'r') as f:
             line = f.readline()
             while line != '':
-
                 if 'SCF Done:' in line:
-                    e_elect = float(line.split()[4]) * constants.E_h * constants.Na
+                    e_elect = float(re.findall(r"SCF Done:  E\(.+\) \=\s+[^\s]+",line)[0].split()[-1])* constants.E_h * constants.Na
```
Comment (hwpang) [inline arkane/ess/gaussian.py:313]:
```suggestion
                    e_elect = float(re.findall(r"SCF Done:  E\(.+\) \=\s+[^\s]+", line)[0].split()[-1]) * constants.E_h * constants.Na
```

File: arkane/ess/gaussian.py [hunk 3]
```diff
@@ -351,7 +350,7 @@ def load_energy(self, zpe_scale_factor=1.):
                     # G4MP2 calculation without opt and freq calculation
                     # Keyword in Gaussian G4MP2(SP), No zero-point or thermal energies are included.
                     e_elect = float(line.split()[2]) * constants.E_h * constants.Na
-
+
```
Comment (hwpang) [inline arkane/ess/gaussian.py:353]:
```suggestion

```

**Review by hwpang — CHANGES_REQUESTED** [review]
LGTM, I only have formatting requests

**Comment by JacksonBurns** [comment]
Rebasing with #2669 now merged (hopefully) fixing CI

**Review by JacksonBurns — DISMISSED** [review]

**Review by hwpang — APPROVED** [review]

**Comment by JacksonBurns** [comment]
Related: https://github.com/dawidd6/action-download-artifact/issues/188#issuecomment-1293565827

**Review by JacksonBurns — APPROVED** [review]

**Pull request merged**